Fix #4181 Fisherman leaving line in water

iiEpic opened the pull request:
Closes #4181 

# Changes proposed in this pull request:
- Checks when sending Citizen to home if the job of the Citizen is Fisherman. If so, retrieveRod is called. Added a null check incase the Fisherman didn't have a line out.

Review please

iiEpic committed:
Fix for #4181

- Checks when sending citizen home to sleep.
- If Job is Fisherman, retrieveRod method is called.

iiEpic committed:
Revert "Fix for #4181"

This reverts commit 98b2fc75e2ec11dc45435039567ccbdd8cf60434.

iiEpic committed:
Fix for #4181

- Checks when sending citizen home to sleep.
- If Job is Fisherman, retrieveRod method is called.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=4652) <br/>All committers have signed the CLA.

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISleep.java:
```diff
@@ -160,6 +162,13 @@ public void tick()
+            if(job.getName() != null) {
+                if (job.getName().contains("Fisherman")) {
+                    EntityAIWorkFisherman fishJob = (EntityAIWorkFisherman) citizen.getCitizenData().getJob().getWorkerAI();
+                    fishJob.retrieveRod();
```
probably better to make this an instanceof JobFisherman

iiEpic replied:
I couldn't quite get that to work because that was my first thing I tried. Let me go at it again and I'll update it! :D

iiEpic committed:
Update to #4181

Changed to instanceOf JobFisherman upon Raycoms request. (Makes alot more sense)

Raycoms committed:
Merge branch 'version/1.15' into version/1.15

Raycoms commented:
I was thinking about about it and I think we might need something better than this. When they go to the restaurant or to the hospital they also gotta call this

iiEpic commented:
That makes sense. Whenever they are not fishing and the bobber is out they need to retrieveRod.

Raycoms commented:
I think we should be able to check in the bobber entity if the fisherman is still fishing atm.

iiEpic commented:
I'll look into it and test it out. Let you know when I got something boss.

iiEpic committed:
Fix for #4181 and others

- Made retrieveRod() back to private
- Removed added code to EntityAISleep
- Added int tickRemove
- Every 100 ticks it will check if the angler's status is set to sleeping, if so it will remove the bobber.

- Thanks to Asher for the code!

iiEpic committed:
Merge branch 'version/1.15' of https://github.com/EpicUnknown/minecolonies into version/1.15

marvin-bitterlich commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/minecolonies/coremod/entity/NewBobberEntity.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=5372724&bid=17612349)

Raycoms reviewed (changes requested):
Almost there.
You don't need the check if the citizen is sleeping.
Instead, every time the fisherman uses the bobber the fisherman should increase the "tickRemove" variable back to 100. 

This way, when he stopped using it, it will automtically hit 0.

(Don't forget to store this tick variable to the entity NBT though)

iiEpic commented:
Got it. I will work on it today

iiEpic committed:
Fix #4181

iiEpic committed:
Fix PathFinding of Fisherman

Fisherman now stops at the edge of the pond instead of trying to fish inside of the pond.

iiEpic committed:
Formatting Changes

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/fisherman/EntityAIWorkFisherman.java:
```diff
@@ -649,11 +649,13 @@ private boolean caughtFish()
-        entityFishHook.remove();
-        worker.getCitizenItemHandler().damageItemInHand(Hand.MAIN_HAND, i);
-        entityFishHook = null;
+        if(entityFishHook != null) {
```
missing

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/NewBobberEntity.java:
```diff
@@ -573,6 +587,8 @@ public boolean isReadyToCatch()
     }
 
+    public void setInUse() { this.tickRemove = 100; }
+
```
add javadoc and remove whitspace (also format)

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/NewBobberEntity.java:
```diff
@@ -143,7 +145,18 @@ public void setPositionAndRotationDirect(double x, double y, double z, float yaw
+                return;
+            }
+        }
+
```
also remove extra whitespace around here

iiEpic committed:
Formatting Changes

I believe I got them all.

Raycoms committed:
Merge branch 'version/1.15' into version/1.15

Raycoms commented:
:shipit: 

[![Approved with PullApprove](https://img.shields.io/badge/maintainers-approved-brightgreen.svg)](https://pullapprove.com/ldtteam/minecolonies/pull-request/4652/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)
